test/views/components/InputSpec.scala

```diff
@@ -75,14 +75,99 @@ class InputSpec extends WordSpec with Matchers with GuiceOneAppPerSuite {
+      private val inputs = doc.getElementsByTag("input").asScala.toList
+      inputs.size shouldBe 1
+      private val inputField = elementAttrs(inputs.head)
+      inputField("value") shouldBe inputValue1
+    }
+
+    "render input with hints" in new Test {
+      private val doc = asDocument(components.input(input, "test", textFormProvider("test" -> nonEmptyText))(fakeRequest, messages, ctx))
+      private val hints = doc.getElementsByTag("span").asScala.toList
+      private val firstHint = hints.head
+      private val hint1Attrs = elementAttrs(firstHint)
+      hint1Attrs("class") shouldBe "govuk-hint"
+      firstHint.text() shouldBe Text(i1Hint).value(messages.lang).head.toString
+    }
+
+    "Input with no body should have a label wrapper class on H1" in new Test {
+      private val doc = asDocument(components.input(inputWithoutBody, "test", textFormProvider("test" -> nonEmptyText))(fakeRequest, messages, ctx))
+      private val h1 = doc.getElementsByTag("h1").first
+      private val attrs = elementAttrs(h1)
+      attrs("class").contains("govuk-label-wrapper") shouldBe true
+    }
+
+    "input without body should render hint within a span without a fieldset" in new Test {
+      private val doc = asDocument(components.input(inputWithHintAndNoBody, "test", textFormProvider("test" -> nonEmptyText))(fakeRequest, messages, ctx))
+      private val fieldset = doc.getElementsByTag("fieldset").first
+
+      Option(fieldset) shouldBe None
+
+      Option(doc.getElementById("input-hint")).fold(fail("Missing hint span")) { span =>
+        val attrs = elementAttrs(span)
+        attrs("id") shouldBe "input-hint"
+        attrs("class").contains("govuk-hint") shouldBe true
+        span.text shouldBe i1Hint(0)
+      }
+    }
+
+   "input with hint in error should include hint id and error id in aria-desribedby on input" in new Test {
+      private val doc = asDocument(components.input(inputWithHintAndErrors, "test", textFormProvider("test" -> nonEmptyText))(fakeRequest, messages, ctx))
+
+      doc.getElementsByTag("input").asScala.toList.foreach { inp =>
+        elementAttrs(inp).get("aria-describedby").fold(fail("Missing aria-describedby")){ aria =>
+          aria should include("input-hint")
+          aria should include("required-error")
+        }
+      }
+   }
+  }
+
+  "English Text Input component" must {
+
+    trait TextTest extends Test {
+      override val input: TextInput = TextInput(Text(i1), Some(Text(i1Hint)), Seq(h2, h3, h4, para1))
+
+      override val inputWithoutBody: TextInput = TextInput(Text(i1), None, Seq.empty)
+      override val inputWithHintAndNoBody: TextInput = TextInput(Text(i1), Some(Text(i1Hint)), Seq.empty)
+      override val inputWithHintAndErrors: TextInput = TextInput(Text(i1), Some(Text(i1Hint)), Seq(para1), Seq(errorMsg))
+      val textInput: TextInput = models.ui.TextInput(Text(), None, Seq.empty)
+      val textInputPage: FormPage = models.ui.FormPage("/url", textInput)
```

Comment:
Do we need this TextInput - can we just use the input defined above?
Can we remove the models.ui. from these definitions

Reply:
They look similar, but the one above uses CurrencyInput as the Input implementation, so itsthe same tests but run through with the TextInput type

Reply:
Sorry, the picture above doesn't show the input I meant - there is a TextInput defined at the top of this section that is used in tests

Reply:
The one above is unused, deleted now